include/dmic.h

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) 2018, Intel Corporation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of the Intel Corporation nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author:	Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+ *		Sathish Kuttan <sathish.k.kuttan@.intel.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef __DMIC_H__
+#define __DMIC_H__
+
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * DMIC driver states
+ */
+enum dmic_state {
+	DMIC_STATE_UNINIT,	/* Uninitialized */
+	DMIC_STATE_CONFIGURED,	/* Configured */
+	DMIC_STATE_READY,	/* Ready */
+	DMIC_STATE_ACTIVE,	/* Active */
+	DMIC_STATE_PAUSED,	/* Paused */
+};
+
+/**
+ * DMIC driver trigger commands
+ */
+enum dmic_trigger {
+	DMIC_TRIGGER_STOP,	/* stop stream */
+	DMIC_TRIGGER_START,	/* start stream */
+	DMIC_TRIGGER_PAUSE,	/* pause the stream */
```

Comment:
What is the difference between PAUSE and STOP?
What is the difference between PAUSED and READY state?

Reply:
CC co-author @singalsu

`PAUSE` is intended to keep the interface running (e.g., for synchronization purposes) but pause any notifications to the applications (e.g., when the load is high and application wants to defer the processing). `RELEASE` is intended to resume the notifications.

The states are
`UNINITIALIZED` --initialize--> `READY`
`READY` --configure-->`CONFIGURED`
`CONFIGURED` --trigger start--> `ACTIVE`
`ACTIVE` --trigger pause--> `PAUSED`
`PAUSED` --trigger resume-->`ACTIVE`
`ACTIVE` --trigger stop--> `CONFIGURED`
`PAUSED` --trigger stop--> `CONFIGURED`

Perhaps, `ACTIVE` should be changed to `INITIALIZED`. Would that be better?

Reply:
IMHO `INITIALIZED` is better name.